cg/meta/demultiplex/validation.py

```diff
@@ -35,9 +36,14 @@ def is_lane_in_fastq_file_name(sample_fastq: Path) -> bool:
     return bool(re.search(r"_L\d+", sample_fastq.name))
 
 
-def is_sample_id_in_directory_name(directory: Path) -> bool:
-    """Validate that directory name contains the sample id formatted as Sample_<sample_id>"""
-    return bool(re.search(r"Sample_\w+", directory.name))
+def is_sample_id_in_directory_name(directory: Path, sample_internal_id: str) -> bool:
+    """Validate that directory name contains the sample id formatted as Sample_<sample_id>."""
+    return bool(re.search(r"Sample_\w+", directory.name)) and sample_internal_id in directory.name
```

Comment:
Still a bit confused over what this condition is doing - would this not match with Sample_dummy_string_<sample_id> as well? Would there be drawbacks with matching "Sample_<sample_id>" or use "return f"Sample_{sample_id}" in directory.name"?

Reply:
No, that is a good point - skipping the regex makes the code more readable